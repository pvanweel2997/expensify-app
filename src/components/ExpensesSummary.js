import React from 'react';
import {connect} from 'react-redux';
import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';


export const ExpensesSummary = (props) => (
    <div>
        {props.expenses.length === 1 ? <p>viewing {props.expenses.length} expense totaling {numeral(props.total / 100).format('$0,0.00')}</p>
        : <p>viewing {props.expenses.length} expenses totaling {numeral(props.total / 100).format('$0,0.00')}</p>
        }
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses,state.filters),
        total: expensesTotal(selectExpenses(state.expenses,state.filters))
    }
};

export default  connect(mapStateToProps)(ExpensesSummary);


