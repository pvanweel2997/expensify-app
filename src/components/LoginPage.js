import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';


export const LoginPage = ({ startLogin }) => (  // startLogin is a function defined in auth.js
    <div>
        <h1>Login</h1>
        <button onClick={startLogin}>Login</button>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined,mapDispatchToProps)(LoginPage);

