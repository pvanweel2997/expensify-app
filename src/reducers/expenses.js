

const expenseReducerDefaultState = [];

export default (state = expenseReducerDefaultState, action) => {
    if (action.type === 'REMOVE_EXPENSE') {
        console.log('action type: '+action.type,action.id)
    }
    switch (action.type) {
        case 'ADD_EXPENSE':
            return [
                ...state, 
                action.expense]
        case 'REMOVE_EXPENSE':
            return state.filter(({id}) => id !== action.id)
        case 'EDIT_EXPENSE':
            return state.map((expense) => {
                if (expense.id === action.id) {
                    return {
                        ...expense,
                        ...action.updates
                    }
                } else {
                    return expense;
                }
            })
        default:
            return state;
    }
}


