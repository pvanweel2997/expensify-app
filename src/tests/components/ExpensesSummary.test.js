import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';


test('Should render ExpensesSummary correctly',() => {
    const wrapper = shallow(<ExpensesSummary  expenses = {expenses} />);
    expect(wrapper).toMatchSnapshot();
})


test('Should render ExpensesSummary correctly',() => {
    const wrapper = shallow(<ExpensesSummary  expenses = {[expenses[0],expenses[2]]} />);
    expect(wrapper).toMatchSnapshot();
})