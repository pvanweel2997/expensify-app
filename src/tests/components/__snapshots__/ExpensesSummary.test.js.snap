// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpensesSummary correctly 1`] = `
<div>
  <p>
    viewing 
    3
     expenses totaling 
    $0.00
  </p>
</div>
`;

exports[`Should render ExpensesSummary correctly 2`] = `
<div>
  <p>
    viewing 
    2
     expenses totaling 
    $0.00
  </p>
</div>
`;
